Automattic/jetpack
# Jetpack Sync: Add Custom table initialization and migration logic

This is a spin-off of #31318 and continuation of #32089 and #32111

## Proposed changes:

This PR adds the Custom table initialization and migration logic.

The PR depends on #32089 and #32111 and needs to have both merged before this one can be merged. We'll also need to update the changelog and package versions afterwards.

Additional details will be added later.

### Other information:

- [x] Have you written new tests for your changes, if applicable? Tests are in #31318

## Jetpack product discussion

p9dueE-7e6-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Apply PR
* Make sure Jetpack Sync works
* Run the Docker PHPUnit tests - `jetpack docker phpunit` and make sure they pass. You can also run only the Sync unit tests with `jetpack docker phpunit -- --testsuite "sync"`
* More tests need to be added to support the switching logic. 
* Manual testing instructions from [PHPUnit tests](https://github.com/Automattic/jetpack/blob/ec40e3e62a94e2b5036adbe4b621930b4d94a572/projects/plugins/jetpack/tests/php/sync/test_class.jetpack-sync-queue-dedicated-table.php#L334) from the Spike PR: #31318
